br/pkg/membuf/limiter.go

```diff
@@ -56,6 +57,10 @@ func (l *Limiter) Release(n int) {
 	l.mu.Lock()
 
 	l.limit += n
+	if l.limit > l.initLimit {
+		panic("limit overflow")
+	}
```

Comment:
when will this happen? can we use other way to handle it, log?

Reply:
I want to fail fast to debug the problem

Reply:
This should not happen, I just suspect a memory problem is caused by it.

Reply:
can we change it to a log with stack, i think it's not a good idea to add panic explicitly for a daemon program except in initialization code